### ISSUE TITLE
/np response different from oppai output (again)

### ISSUE THREAD

**Issue by omkelderman** [Tillerino/Tillerinobot#96]
cuz reasons i wanted to post the pp-value of Knife Party - Centipede in #general on discord server and once again i saw a difference between the /np response and running raw oppai

/np:
![](https://omk.pics/1/yaffj.png)

oppai:
![](https://omk.pics/1/UaTKB.png)

with a difference of about 40pp, thats a lot to me, but ofc this being a ridiculously high number maybe its not that much, idk...
Either way this is (obviously) an unranked map, so it should match oppai output regardless right?

**Comment by Tillerino** [comment]
This probably has something to do with the fact that oppai rounds all its
output.

Oppai gives two decimals for aim and speed stars. So the relative error
here (numbers around 20) is around 1E-4. 40pp difference for a number of
157430 is also a relative error around 1E-4.

I can try upping the JSON output precision in my oppai fork as I did with
the non-JSON output before.

**Comment by omkelderman** [comment]
well, there's https://github.com/Francesco149/oppai/commit/2ec7b4be91b7aa3124058b7b0f0a0db14d660e41 dunno if that helps (screenshot is with oppai version including that commit)

**Comment by omkelderman** [comment]
oh wait, that shouldnt matter for the screenshot ofc xd, the rounding is added back in text-output-mode. Json output (with that commit) gives this (newlines added by me for readability):

```
{
"oppai_version":"0.6.2",
"artist":"Knife Party",
"title":"Centipede",
"version":"This isn't a map, just a simple visualisation",
"creator":"Sugoi-_-Desu",
"mods_str": "",
"od":0,
"ar":10,
"cs":5,
"combo": 6368,
"max_combo":6368,
"num_circles": 6356,
"num_sliders": 2,
"num_spinners": 3,
"misses":0,
"score_version":1,
"stars": 48.01,
"speed_stars": 17.47,
"aim_stars": 26.18,
"pp":157430
}
```

aim and speed stars are still two decimals eventho it shouldnt because that commit? So bug in oppai? idk....

**Comment by Tillerino** [comment]
Correct, it's the output that does the rounding.

**Comment by omkelderman** [comment]
well, in theory the json-output should not do any rounding if i understand that latest commit correctly, i'll take a look at it

**Comment by omkelderman** [comment]
ok, this is getting weird:

* the above tests are ran by downloading https://github.com/Francesco149/oppai/releases/latest
* that release has name "0.6.2" as also seen in the screenshit, BUT on that page it says it includes the latest commit, which includes the change of no rounding AND upping the version to 0.6.3.
* if i download the source myself and compile it myself, and thus running 0.6.3 it gives the same result as the /np result, as expected....

tl;dr i thought the oppai build i was using on those results was including the "do not round anymore" commit, but in fact **it was not, eventho it says it is, but it also kinda says it isnt**

tl;dr2: shit is confusing as hell
